pkg/monitor/monitor/util.go

```diff
@@ -319,7 +331,118 @@ chmod 777 /data/prometheus /data/grafana
 	return container
 }
 
-func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster) core.Container {
+func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMCluster) core.Container {
+	c := `mkdir -p /data/prometheus
+chmod 777 /data/prometheus
+/usr/bin/init.sh`
+	if monitor.Spec.Grafana != nil {
+		c = `mkdir -p /data/prometheus /data/grafana
+chmod 777 /data/prometheus /data/grafana
+/usr/bin/init.sh`
+	}
+	command := []string{
+		"/bin/sh",
+		"-c",
+		c,
+	}
+	alertManagerRulesVersion := dc.MasterImage()
+	if monitor.Spec.AlertManagerRulesVersion != nil {
+		alertManagerRulesVersion = fmt.Sprintf("dm:%s", *monitor.Spec.AlertManagerRulesVersion)
+	}
+	container := core.Container{
+		Name:  "monitor-dm-initializer",
+		Image: fmt.Sprintf("%s:%s", monitor.Spec.DMInitializer.BaseImage, monitor.Spec.DMInitializer.Version),
+		Env: []core.EnvVar{
+			{
+				Name:  "DM_CLUSTER_NAME",
+				Value: dc.Name,
+			},
+			{
+				Name:  "PROM_CONFIG_PATH",
+				Value: "/prometheus-rules",
+			},
+			{
+				Name:  "PROM_PERSISTENT_DIR",
+				Value: "/data",
+			},
+			{
+				Name:  "DM_VERSION",
+				Value: alertManagerRulesVersion,
+			},
+			{
+				Name:  "GF_DM_PROMETHEUS_URL",
+				Value: "http://127.0.0.1:9090",
+			},
+			{
+				Name:  "DM_CLUSTER_NAMESPACE",
+				Value: dc.Namespace,
+			},
+			{
+				Name:  "TZ",
+				Value: dc.Spec.Timezone,
+			},
+		},
+		Command: command,
+		VolumeMounts: []core.VolumeMount{
+			{
+				MountPath: "/prometheus-rules",
+				Name:      "prometheus-rules",
+				ReadOnly:  false,
+			},
+			{
+				MountPath: "/data",
+				Name:      "monitor-data",
+			},
+		},
+		Resources: controller.ContainerResource(monitor.Spec.DMInitializer.ResourceRequirements),
+	}
+
+	if monitor.Spec.DMInitializer.ImagePullPolicy != nil {
+		container.ImagePullPolicy = *monitor.Spec.DMInitializer.ImagePullPolicy
+	}
+
+	if monitor.Spec.KubePrometheusURL != nil {
+		container.Env = append(container.Env, core.EnvVar{
+			Name:  "GF_K8S_PROMETHEUS_URL",
```

Comment:
What this env used for?

Reply:
Configure k8s datasource url in https://github.com/pingcap/monitoring/blob/master/platform-monitoring/operator/init.sh#L69

Reply:
That's because the other json files have some dashboards that need to retrieve metrics from the Prom for Kubernetes, any dashboards from the dm json file have the same requirements?

Reply:
DM do not need this env, i will remove it when building dm-monitor-initializer container.